pyproject.toml

```diff
@@ -1,7 +1,70 @@
 [build-system]
-requires = ["setuptools", "lsst-versions"]
+requires = ["setuptools", "lsst-versions >= 1.3.0"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "lsst-daf-butler"
+description = "An abstraction layer for reading and writing astronomical data to datastores."
+license = {text = "GPLv3+ License"}
+readme = "README.md"
+authors = [
+    {name="Rubin Observatory Data Management", email="dm-admin@lists.lsst.org"},
+]
+classifiers = [
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Topic :: Scientific/Engineering :: Astronomy",
+]
+keywords = ["lsst"]
+dependencies = [
+    "astropy >=4.0",
+    "pyyaml >=5.1",
+    "sqlalchemy >= 1.4",
+    "click >= 7.0",
+    "lsst-sphgeom",
+    "lsst-utils",
+    "lsst-resources",
+    "deprecated >= 1.2",
+    "pydantic",
+]
+
+dynamic = ["version"]
+[project.urls]
+"Homepage" = "https://github.com/lsst/daf_butler"
+[project.optional-dependencies]
+test = [
```

Comment:
In theory you can do DM-35272 (add a postgres option here) since you are editing the file but doing it on a separate ticket is fine.

Reply:
Implementing this in  DM-35272